torch/_meta_registrations.py

```diff
@@ -3646,6 +3646,50 @@ def meta_relu_(self):
 
 @register_meta([aten.index_put.default, aten._unsafe_index_put.default])
 def meta_index_put(self, indices, values, accumulate=False):
+    if not isinstance(indices, tuple):
+        indices = (indices,)
+
+    torch._check(
+        len(indices) > 0,
+        lambda: "Passing an empty index list to torch.index_put() is not valid syntax",
+    )
+    torch._check(
+        self.device == values.device,
+        lambda: "Tensor and value must be on the same device",
+    )
+
+    for tensor in indices:
+        if tensor is not None:
+            dtype = tensor.dtype
+            torch._check(
+                dtype in [torch.long, torch.int64, torch.bool, torch.uint8],
+                lambda: "tensors used as indices must be long, int, byte or bool tensors",
+            )
+
+    torch._check(
+        self.dim() >= len(indices),
+        lambda: f"too many indices for tensor of dimension {self.dim()} (got {len(indices)})",
+    )
+    torch._check(
+        not indices.empty or utils.is_expandable_to(values.size(), self.size()),
+        lambda: (
+            f"shape mismatch: value tensor of shape {values.size()} "
+            f"cannot be broadcast to indexing result of shape {self.size()}"
+        ),
+    )
+    torch._check(
+        not accumulate,
+        lambda: "index_put does not support accumulate=true",
+    )
```

Comment:
Where is this coming from?
```python
>>> torch.index_put(torch.randn(3), [torch.tensor([1,2])], torch.randn(2), accumulate=True)
tensor([0.0319, 0.3860, 1.6362])
```

In general, each check should be mimicking the checks in eager. I don't think these are the checks we have in eager.

Reply:
I see... I implemented that case following this https://github.com/pytorch/pytorch/blob/ead97ee4863253b37e03e4d14a71f19d435db4cf/aten/src/ATen/native/cuda/IndexKernel.cu#L237

I believe we should update this, too. Because this works:

```python
In [2]: torch.index_put(torch.randn(3, device='cuda'), [torch.tensor([1,2], device='cuda')], torch.randn(2, device='cuda'), accumulate=True)
Out[2]: tensor([0.6225, 1.1404, 0.6779], device='cuda:0')
```

Also, I see I made a couple of errors here; I'll update them soon and ping you. Thank you!